controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_types.go

```diff
@@ -234,6 +239,20 @@ type RemediationStrategy struct {
 	MinHealthyPeriod *metav1.Duration `json:"minHealthyPeriod,omitempty"`
 }
 
+// MachineNamingStrategy allows changing the naming pattern used when creating Machines.
+// InfraMachines & KubeadmConfigs will use the same name as the corresponding Machines.
+type MachineNamingStrategy struct {
+	// Template defines the template to use for generating the names of the Machine objects.
+	// If not defined, it will fallback to `{{ .kubeadmcontrolplane.name }}-{{ .random }}`.
+	// If the templated string exceeds 63 characters, it will be trimmed to 58 characters and will
+	// get concatenated with a random suffix of length 5.
+	// The templating mechanism provides the following arguments:
+	// * `.kubeadmcontrolplane.name`: The name of the KubeadmControlPlane object.
+	// * `.random`: A random alphanumeric string, without vowels, of length 5.
+	// +optional
+	Template string `json:"template,omitempty"`
```

Comment:
Is it possible to impose a pattern validation on this string?

Is there any admission time validation at present that checks that this compiles and would produce a valid template?

Reply:
Good catch!
(I didn't review the implementation myself yet, only the API)

We should have a similar validation than what we have for other naming strategies: https://github.com/kubernetes-sigs/cluster-api/blob/8752f4f91301b970c8f50f2bb9c39f5a50caea5d/internal/webhooks/clusterclass.go#L437

Reply:
I have added the validation webhook, please check.